commit b6ab149654ef1fada0b37b379c04c475c2fdc675
Author: Horatiu Vultur <horatiu.vultur@microchip.com>
Date:   Tue Jan 25 12:48:15 2022 +0100

    net: lan966x: Fix sleep in atomic context when injecting frames
    
    On lan966x, when injecting a frame it was polling the register
    QS_INJ_STATUS to see if it can continue with the injection of the frame.
    The problem was that it was using readx_poll_timeout which could sleep
    in atomic context.
    This patch fixes this issue by using readx_poll_timeout_atomic.
    
    Fixes: d28d6d2e37d10d ("net: lan966x: add port module support")
    Signed-off-by: Horatiu Vultur <horatiu.vultur@microchip.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/drivers/net/ethernet/microchip/lan966x/lan966x_main.c b/drivers/net/ethernet/microchip/lan966x/lan966x_main.c
index 2cb70da63db3..1f60fd125a1d 100644
--- a/drivers/net/ethernet/microchip/lan966x/lan966x_main.c
+++ b/drivers/net/ethernet/microchip/lan966x/lan966x_main.c
@@ -182,9 +182,9 @@ static int lan966x_port_inj_ready(struct lan966x *lan966x, u8 grp)
 {
 	u32 val;
 
-	return readx_poll_timeout(lan966x_port_inj_status, lan966x, val,
-				  QS_INJ_STATUS_FIFO_RDY_GET(val) & BIT(grp),
-				  READL_SLEEP_US, READL_TIMEOUT_US);
+	return readx_poll_timeout_atomic(lan966x_port_inj_status, lan966x, val,
+					 QS_INJ_STATUS_FIFO_RDY_GET(val) & BIT(grp),
+					 READL_SLEEP_US, READL_TIMEOUT_US);
 }
 
 static int lan966x_port_ifh_xmit(struct sk_buff *skb,